src/main/kotlin/org/ageseries/libage/sim/thermal/Thermal.kt

```diff
@@ -0,0 +1,306 @@
+    /** Given a temperature [t] in this unit, return its value in Kelvin. */
+    fun unmap(t: Double): Double = (t - base) / factor
+
+    /**
+     * Give a human-readable string of a temperature in this scale.
+     */
+    fun display(t: Double): String = "${t}${displayUnit}"
+
+    companion object {
+        /**
+         * Kelvin; absolute zero, with steps of a hundredth of a grade.
+         *
+         * Useless directly, but maybe useful as an argument where a Scale is expected.
+         */
+        val KELVIN = Scale(1.0, 0.0, "K")
+        /**
+         * Absolute Grade; absolute zero, with steps equal to the grade.
+         */
+        val ABSOLUTE_GRADE = Scale(0.01, 0.0, "AG")
+        /**
+         * Rankine; absolute zero, with steps one-hundred-eightieth of a grade.
+         */
+        val RANKINE = Scale(9.0/5.0, 0.0, "°R")
+        /**
+         * Grade; zero is pure water freezing, one is pure water boiling, all at standard pressure.
+         */
+        val GRADE = Scale(0.01, -2.7315, "G")
+        /**
+         * Centigrade; zero as Grade, but with steps one-hundredth of a grade.
+         */
+        val CENTIGRADE = Scale(1.0, -273.15, "°C")
+        /**
+         * Celsius, an alias for Centigrade.
+         */
+        val CELSIUS = CENTIGRADE
+        /**
+         * Milligrade; zero as Grade, but with steps one-thousandth of a grade.
+         */
+        val MILLIGRADE = Scale(10.0, -2731.5, "mG")
+        /**
+         * Fahrenheit; 32 is zero grade, 212 is one grade.
+         */
+        val FAHRENHEIT = Scale(9.0/5.0, -491.67, "°F")
+    }
+}
+/**
+ * A temperature.
+ *
+ * The inner type is always Kelvin; conversions are available as properties.
+ */
+@JvmInline
+value class Temperature(val kelvin: Double) {
+    /**
+     * Return this temperature on the given [Scale].
+     */
+    fun in_(scale: Scale): Double = scale.map(kelvin)
+
+    operator fun plus(rhs: Temperature) = Temperature(kelvin + rhs.kelvin)
+    operator fun minus(rhs: Temperature) = Temperature(kelvin - rhs.kelvin)
+
+    operator fun compareTo(rhs: Temperature) = kelvin.compareTo(rhs.kelvin)
+
+    override fun toString() = Scale.KELVIN.display(kelvin)
+
+    companion object {
+        /**
+         * Return a Temperature given a measurement on a different [Scale].
+         */
+        fun from(temp: Double, scale: Scale) = Temperature(scale.unmap(temp))
+    }
+}
+/**
+ * "Standard" temperature, in Kelvin.
+ *
+ * This is used as the default temperature for thermal masses.
+ */
+val STANDARD_TEMPERATURE: Temperature = Temperature(273.15)
+
+/**
+ * A description of a material's instrinsic thermal properties.
+ *
+ * You shouldn't need to make one of these unless you're really describing a new material; otherwise,
+ * the members here cover sane defaults.
+ */
+class Material(
+    /** Specific heat capacity, in J/Kkg. */
+    val specificHeat: Double,
+    /** Intrinsic conductance, in W/mK. */
+    val conductance: Double,
```

Comment:
Thermic or electric conductance? Be specific.

Reply:
We _are_ in the `thermal` package, which is also why I was playing fast and loose with other names, like `Connection` and `Simulator` :)

~~At least it's not `space.Object` :P~~

Reply:
On review, in fact, I feel like I didn't go far enough: _since this is `thermal`_, some of these classes might just be `Mass`, `Connection`, `ConnectionParameters`, etc.

Reply:
This is in thermal, but the issue is that Materials is extensible beyond thermal.